lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java

```diff
@@ -210,12 +216,27 @@ public IndexSearcher(IndexReader r, Executor executor) {
   public IndexSearcher(IndexReaderContext context, Executor executor) {
     assert context.isTopLevel: "IndexSearcher's ReaderContext must be topLevel for reader" + context.reader();
     reader = context.reader();
-    this.executor = executor;
+    this.sliceExecutionControlPlane = getSliceExecutionControlPlane(executor);
     this.readerContext = context;
     leafContexts = context.leaves();
     this.leafSlices = executor == null ? null : slices(leafContexts);
   }
 
+  /**
+   * We do this elaborate dance as to have only one constructor with a nullable second parameter
+   * See the next constructor for more clarification
+   * Only for testing
+   */
+  IndexSearcher(IndexReaderContext context, Executor executor,
```

Comment:
this doesn't need an executor, does it?

Reply:
It is needed since executor is a final member of IndexSearcher hence needs to be initialized? We could potentially always set it to null since it anyways wont be used but that seemed a bit counter intuitive. WDYT?